sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/write/WriteDocumentsService.java

```diff
@@ -0,0 +1,435 @@
+   */
+  private Uni<DocumentResponseWrapper<Void>> patchDocumentInternal(
+      String namespace,
+      String collection,
+      String documentId,
+      List<String> subPath,
+      String payload,
+      Integer ttl,
+      ExecutionContext context) {
+    // pre-process to support array elements
+    final List<String> subPathProcessed = processSubDocumentPath(subPath);
+    // read the root
+    final JsonNode root = readPayload(payload);
+    // explicitly forbid arrays and empty objects
+    if (root.isArray()) {
+      throw new ErrorCodeRuntimeException(ErrorCode.DOCS_API_PATCH_ARRAY_NOT_ACCEPTED);
+    }
+    if (root.isObject() && root.isEmpty()) {
+      throw new ErrorCodeRuntimeException(ErrorCode.DOCS_API_PATCH_EMPTY_NOT_ACCEPTED);
+    }
+
+    return jsonSchemaManager
+        .validateJsonDocument(
+            tableManager.getValidCollectionTable(namespace, collection),
+            root,
+            !subPathProcessed.isEmpty())
+        .onItem()
+        .transform(
+            __ -> {
+              // shred rows
+              List<JsonShreddedRow> rows = documentShredder.shred(root, subPathProcessed);
+
+              // call patch document
+              return writeBridgeService.patchDocument(
+                  namespace, collection, documentId, subPathProcessed, rows, ttl, context);
+            })
+        .map(any -> new DocumentResponseWrapper<>(documentId, null, null, context.toProfile()));
+  }
+
+  /**
+   * Deletes a document with given ID in the given namespace and collection at the specified
+   * sub-path. Any previously existing sub-document at the given path will be removed.
+   *
+   * @param namespace Namespace
+   * @param collection Collection name
+   * @param documentId The ID of the document to delete
+   * @param subPath Sub-path of the document to delete. If empty will delete the whole doc.
+   * @param context Execution content
+   * @return Flag representing if the operation was success.
+   */
+  public Uni<Boolean> deleteDocument(
+      String namespace,
+      String collection,
+      String documentId,
+      List<String> subPath,
+      ExecutionContext context) {
+    // pre-process to support array elements
+    List<String> subPathProcessed = processSubDocumentPath(subPath);
+    return writeBridgeService
+        .deleteDocument(namespace, collection, documentId, subPathProcessed, context)
+        .onItem()
+        .transform(__ -> true);
+  }
+
+  // we need to transform the stuff to support array elements
+  private List<String> processSubDocumentPath(List<String> subDocumentPath) {
+    return subDocumentPath.stream()
+        .map(path -> DocsApiUtils.convertArrayPath(path, configuration.maxArrayLength()))
+        .collect(Collectors.toList());
+  }
+
+  private Uni<Integer> determineTtl(
+      String namespace, String collection, String documentId, ExecutionContext ctx) {
+    return readBridgeService
+        .getDocumentTtlInfo(namespace, collection, documentId, ctx)
```

Comment:
hmm, here was a mistake when creating ` .getDocumentTtlInfo`, it should have been a Uni<RawDocument> here, as there is no chance we would emit more than one document..

can you fix it there by adding:

```
.select().first().toUni(); 
```

to the end of chain in the read service and then you will get only one element here and don't have to collect list and do any magic of selecting only the first document
```

Reply:
It's also fine to leave the Read service as is and do the stuff directly here.. as if multi does not emit you can have a switch..